### PR TITLE
Update platforms to modern versions

### PR DESCRIPTION
**Pull request by ramereth**
- Add support for AlmaLinux, Rocky Linux and Fedora
- Update and fixes tests

Signed-off-by: Lance Albertson <lance@osuosl.org>

# Description

Describe what this change achieves

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [ ] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by ramereth** [comment]
@bmhughes I know you use this cookbook, let me know if you have an issue with my apparmor fix

File: libraries/helpers.rb [hunk 1]
```diff
@@ -112,25 +112,28 @@ def dhcpd_lib_dir
...
-            'group' => 'dhcpd',
-            'mode' => '0775',
-          }
         when 'debian'
-          {
-            'owner' => 'root',
-            'group' => 'root',
-            'mode' => '0755',
-          }
+          case node['platform']
```
Comment (bmhughes) [inline libraries/helpers.rb:123]: On balance I think this was better as it was before without the nested `case`

File: libraries/helpers.rb [hunk 2]
```diff
@@ -146,28 +149,31 @@ def dhcpd_lease_file(ip_version)
...
-            'action' => :create_if_missing,
-          }
         when 'debian'
-          {
-            'owner' => 'root',
-            'group' => 'root',
-            'mode' => '0644',
-            'action' => :create_if_missing,
-          }
+          case node['platform']
```
Comment (bmhughes) [inline libraries/helpers.rb:161]: Ditto above

File: resources/service.rb [hunk 3]
```diff
@@ -96,6 +96,11 @@ def do_service_action(resource_action)
 
       action :create
     end if dhcpd_use_systemd?
+
+    # Work around issue with apparmor on Ubuntu
```
[the remark this answers is not in the document]
Comment (bmhughes) [reply resources/service.rb:100]: That sounds good to me.

File: libraries/helpers.rb (hunk 1 above)
Comment (bmhughes) [reply libraries/helpers.rb:123]: Ah I follow you, probably the better option then.

**Comment by kitchen-porter** [comment]
Released as: [8.1.5](https://github.com/sous-chefs/dhcp/releases/tag/8.1.5)

**Pull request merged**